.github/workflows/goreleaser.yaml

```diff
@@ -0,0 +1,56 @@
+name: release
+on:
+  pull_request:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
+
+      - name: Get the image tag
+        run: |
+          # Source: https://github.community/t/how-to-get-just-the-tag-name/16241/32
+          if [[ $GITHUB_REF == refs/tags/* ]]; then
+            echo IMAGE_TAG="${GITHUB_REF#refs/tags/}" >> $GITHUB_ENV
+          fi
+
+      - name: Docker Login
+        uses: docker/login-action@v1
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_PASSWORD }}
+
+      - name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v2
+        with:
+          version: latest
+          args: release --rm-dist
+        env:
+          GITHUB_TOKEN: ${{ secrets.GORELEASER_GITHUB_TOKEN }}
+          IMAGE_REPO: ${{ secrets.QUAY_USERNAME }}/olm
+          PKG: github.com/operator-framework/operator-lifecycle-manager
+
+      - name: Generate quickstart release manifests
+        run: make release ver=${{ env.IMAGE_TAG }} IMAGE_REPO=quay.io/${{ secrets.QUAY_USERNAME }}/olm
+
+      - name: Update release artifacts with rendered Kubernetes manifests
+        uses: softprops/action-gh-release@v1
+        with:
+          name: ${{ env.IMAGE_TAG }}
+          files: |
+            deploy/upstream/quickstart/crds.yaml
+            deploy/upstream/quickstart/olm.yaml
+          draft: true
+          token: ${{ secrets.GORELEASER_GITHUB_TOKEN }}
```

Comment:
GoReleaser has some ability to create GH releases and upload artifacts there. You may be able to move the release manifest generation above (or maybe included in) the goreleaser configuration and then have goreleaser handle creating the release and uploading artifacts.

Side note: only reason we're not doing this in operator-registry is because GoReleaser can't build a macOS binary alongside linux binaries without a cross-compiler running, and Apple doesn't permit that in general.

Reply:
Unfortunately, I've already tried rendering these release manifests before running the goreleaser action, but it dirtied the local environment (which I assume maps to `git diff` produces output) and failed the release process. I might be able to instead migrate towards generating these manifests, uploading using the archiving action, restore the git environment, and someone attach these files using the `extra_files` stanza, but it's not immediately clear how that'd work if those files are achived and not present locally. Maybe a .gitignore could work?

Another implementation that I played around with trying to global after hooks but that's a pro feature and there wasn't a step between building and pushing the manifestlist images and building out the release.

Reply:
Maybe the cleaner option would be to create a draft release first, then run goreleaser, then upload any post-goreleaser artifacts that need to be attached so it's clear we're creating the release outside of goreleaser (and thus increase readability)? That's assuming goreleaser has support for working with existing draft releases, which playing around in my own fork seems to be a safe assumption.

Reply:
Yeah perhaps separating them is cleaner in this case. It seems like getting things _just so_ for goreleaser would just add complexity and obfuscation for no real gain.

Reply:
Ended up adding an action that's responsible for creating a draft release and it looks like goreleaser plays well with updating an existing draft release. So the overall workflow would be create an empty draft releaser -> run goreleaser to produce multi-arch container images and manifestlist images, and a generated changelog -> add post-goreleaser releaser artifacts by generating the quickstart manifests and using an action to attach those artifacts to this existing draft release.

The first step obviously isn't needed, but I think it improves the readability of the workflow file quite a bit as goreleaser abstracts quite a bit, and removes the multi-stage release process and removes the need for housing rendered YAML release manifests entirely. This means the new release process is maintainer creates a new tag locally and pushes that upstream -> the release action runs and creates a draft release -> maintainer publishes that draft release.